### PR TITLE
chore: Use en-US localization as a fallback if current language does not have a translation

### PR DESCRIPTION
**Pull request by kaphacius**
This PR adds a fallback localization mechanism. In case current language does not have a translation for a specific key, an English translation will be used. We consider en-US `Localizable.strings` to be the most complete version.

### CODE REVIEW

File: Demo/Configuration.swift [hunk 1]
```diff
@@ -18,11 +18,11 @@ internal enum Configuration {
 
     static let appName = "Adyen Demo"
 
-    static let amount = Payment.Amount(value: 17408, currencyCode: "EUR")
+    static let amount = Payment.Amount(value: 17408, currencyCode: "BRL")
 
     static let reference = "Test Order Reference - iOS UIHost"
     
-    static let countryCode = "NL"
+    static let countryCode = "BR"
```
Comment (descorp) [inline Demo/Configuration.swift:25]: don't commit this. It also breaks Swift UI tests

File: Adyen/Utilities/Localization.swift [hunk 2]
```diff
@@ -41,7 +42,16 @@ public func localizedString(_ key: LocalizationKey, _ parameters: LocalizationPa
 }
 
 private func fallbackLocalizedString(key: String) -> String {
-    NSLocalizedString(key, tableName: nil, bundle: Bundle.coreInternalResources, comment: "")
+    let localizedFallback = NSLocalizedString(key, tableName: nil, bundle: Bundle.coreInternalResources, comment: "")
+    
+    if localizedFallback != key && localizedFallback.isEmpty == false {
```
Comment (descorp) [inline Adyen/Utilities/Localization.swift:47]: It can't be equals `key`, right? 🤔

Comment (descorp) [reply Adyen/Utilities/Localization.swift:47]: Why can add `var nonEmpty: Bool ` extension for String 😆

Comment (kaphacius) [reply Adyen/Utilities/Localization.swift:47]: if the key is missing from the `Localizable.strings`, `NSLocalizedString` will return the key. here we need to check for that

File: Demo/Configuration.swift (hunk 1 above)
Comment (kaphacius) [reply Demo/Configuration.swift:25]: 👍

File: Adyen/Utilities/Localization.swift [hunk 3]
```diff
@@ -41,7 +42,16 @@ public func localizedString(_ key: LocalizationKey, _ parameters: LocalizationPa
...
+        return localizedFallback
+    } else {
+    // Fallback to en-US
+        return Bundle.coreInternalResources.path(forResource: "en-US", ofType: "lproj")
+            .flatMap(Bundle.init(path: ))
+            .map { NSLocalizedString(key, tableName: nil, bundle: $0, comment: "") } ?? key
```
Comment (mohammedDehairy) [inline Adyen/Utilities/Localization.swift:53]: why not put this in the `buildPossibleInputs`?

Comment (kaphacius) [reply Adyen/Utilities/Localization.swift:53]: this is more of a fallback option, so i thought it should be in `fallbackLocalizedString`

File: Adyen/Utilities/Localization.swift (hunk 2 above)
Comment (mohammedDehairy) [reply Adyen/Utilities/Localization.swift:47]: yeah, actually you're right, here is better.

but can we unit test this in `ADYLocalizationTests`?

Comment (kaphacius) [reply Adyen/Utilities/Localization.swift:47]: will do! 👍

File: Adyen/Utilities/Localization.swift (hunk 3 above)
Comment (mohammedDehairy) [reply Adyen/Utilities/Localization.swift:53]: yeah, actually you're right, here is better.

but can we unit test this in `ADYLocalizationTests`?

Comment (kaphacius) [reply Adyen/Utilities/Localization.swift:53]: unfortunately i have not found a reliable way to test this

Comment (mohammedDehairy) [reply Adyen/Utilities/Localization.swift:53]: 👍 no problem.

**Review by mohammedDehairy — APPROVED** [review]

**Pull request merged**